python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2132,6 +2143,14 @@ def get_tick_retention_settings(
         default_tick_settings = get_default_tick_retention_settings(instigator_type)
         return get_tick_retention_settings(tick_settings, default_tick_settings)
 
+    def inject_env_vars(self, location_name: Optional[str]):
+        if not self._secrets_loader:
+            return
+
+        new_env = self._secrets_loader.load_env_vars(location_name)
+        for k, v in new_env.items():
+            os.environ[k] = v
```

Comment:
should we log what keys are adding/changing values? engine event + metadata?

Reply:
interesting idea in the run case (there isn't an engine event in the grpc server case)